src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1650,20 +1651,36 @@ MavlinkReceiver::handle_message_play_tune(mavlink_message_t *msg)
 	mavlink_play_tune_t play_tune;
 	mavlink_msg_play_tune_decode(msg, &play_tune);
 
-	char *tune = play_tune.tune;
+	if ((mavlink_system.sysid == play_tune.target_system || play_tune.target_system == 0) &&
+	    (mavlink_system.compid == play_tune.target_component || play_tune.target_component == 0)) {
 
-	if ((mavlink_system.sysid == play_tune.target_system ||
-	     play_tune.target_system == 0) &&
-	    (mavlink_system.compid == play_tune.target_component ||
-	     play_tune.target_component == 0)) {
+		Tunes tunes;
 
-		if (*tune == 'M') {
-			int fd = px4_open(TONE_ALARM0_DEVICE_PATH, PX4_F_WRONLY);
+		tune_control_s tune_control {};
+		tune_control.tune_id = 0;
+		tune_control.volume = tune_control_s::VOLUME_LEVEL_DEFAULT;
 
-			if (fd >= 0) {
-				px4_write(fd, tune, strlen(tune) + 1);
-				px4_close(fd);
-			}
+		// Let's make sure the input is 0 terminated and we don't ever overrun it.
+		play_tune.tune2[sizeof(play_tune.tune2) - 1] = '\0';
+
+		tunes.set_string(play_tune.tune, tune_control.volume);
+
+		unsigned frequency;
+		unsigned duration;
+		unsigned silence;
+		uint8_t volume;
+
+		while (tunes.get_next_note(frequency, duration, silence, volume) > 0) {
+			tune_control.tune_id = 0;
+			tune_control.frequency = (uint16_t)frequency;
+			tune_control.duration = (uint32_t)duration;
+			tune_control.silence = (uint32_t)silence;
+			tune_control.volume = (uint8_t)volume;
```

Comment:
Hi @julianoes , would you want to use `static_cast<type>(var)` , (C++ style casts), here?  (It allows compile time type checking where C-style casting does not.)